package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -539,8 +539,13 @@ dhcp_add() {
 	# Do not support non-static interfaces for now
 	[ static = "$proto" ] || return 0
 
+	ipaddr="${subnet%%/*}"
+	prefix_or_netmask="${subnet##*/}"
+
 	# Override interface netmask with dhcp config if applicable
-	config_get netmask "$cfg" netmask "${subnet##*/}"
+	config_get netmask "$cfg" netmask
+
+	[ -n "$netmask" ] && prefix_or_netmask="$netmask"
```

Comment:
You can optimize this part:
```sh
# Override interface netmask with dhcp config if applicable
config_get netmask "$cfg" netmask

ipaddr="${subnet%%/*}"
netmask_or_prefix="${netmask:-${subnet##*/}}"
```
And use `netmask_or_prefix` later as it looks more logical.